src/content/en/updates/posts/2016/05/houdini.md

```diff
@@ -0,0 +1,245 @@
+
+{% highlight JS %}
+class {
+  static get inputProperties() {
+    return ['border-color', 'border-size'];
+  }
+  paint(ctx, geom, inputProperties) {
+    var offset = inputProperties['border-size']
+    var colors = inputProperties['border-color'];
+    self.drawFadingEdge(
+      ctx,
+      0-offset[0], 0-offset[0],
+      geom.width+offset[0], 0-offset[0],
+      color[0]);
+    self.drawFadingEdge(
+      ctx,
+      geom.width+offset[1], 0-offset[1],
+      geom.width+offset[1], geom.height+offset[1],
+      color[1]);
+    self.drawFadingEdge(
+      ctx, 0-offset[2],
+      geom.height+offset[2], geom.width+offset[2],
+      geom.height+offset[2],
+      color[2]);
+    self.drawFadingEdge(
+      ctx,
+      0-offset[3], 0-offset[3],
+      0-offset[3], geom.height+offset[3],
+      color[3]);
+  }
+  drawFadingEdge(ctx, x0, y0, x1, y1, color) {
+    var gradient =
+      ctx.createLinearGradient(x0, y0, x1, y1);
+    gradient.addColorStop(0, color);
+    var colorCopy = new ColorValue(color);
+    colorCopy.opacity = 0;
+    gradient.addColorStop(0.5, colorCopy);
+    gradient.addColorStop(1, color);
+  }
+  overflow(inputProperties) {
+    // Taking a wild guess here. The return type
+    // of overflow() is currently specified
+    // as `void`, lol.
+    return {
+      top: inputProperties['border-size'][0],
+      right: inputProperties['border-size'][1],
+      bottom: inputProperties['border-size'][2],
+      left: inputProperties['border-size'][3],
+    };
+  }
+};
+{% endhighlight %}
+
+### Compositor Worklet
+
+At the time of writing, the compositor worklet doesn’t even have a proper draft and yet it’s the one that gets me excited the most. As you might know, some operations are outsourced to the graphics card of your computer by the CSS engine, although that is dependent on both your grapics card and your device in general. A browser usually takes the DOM tree and, based on specific criteria, decides to give some branches and subtrees their own [layer][HTML5Rocks layers]. These subtrees paint themselves onto it (maybe using a paint worklet in the future). As a final step, all these individual, now painted, layers are stacked and positioned on top of each other, respecting z-indices, 3D transforms and such, to yield the final image that is visible on your screen. This process is called “compositing” and is executed by the “compositor”. The advantage of this process is that you don’t have to make *all* the elements repaint themselves when you scrolled a tiny bit. Instead, you can reuse the layers from the previous frame and just re-run the compositor with the changed scroll position. This makes things fast. This makes us reach 60fps. This makes [Paul Lewis] happy.
```

Comment:
There are a bunch of `[]` content that isn't linked to anything, eg `This makes [Paul Lewis] happy.` Is that intentional?